modyn/trainer_server/internal/dataset/online_dataset.py

```diff
@@ -409,8 +626,45 @@ def _fetch_partition_noprefetch(
                 container["weights"][idx],
             )
 
+    def _fetch_partition_noprefetch_generative(  # based on the one above
```

Comment:
similar comments here. I would propose to refactor the PR such that we don't have parallel codepaths for generative